ansible/roles/gpu/tasks/nvidia-gpu-CentOS7.yaml

```diff
@@ -0,0 +1,91 @@
+---
+- name: install epel-release
+  yum:
+    name: epel-release
+    state: present
+    update_cache: true
+
+- name: find this kernel headers
+  yum:
+    list: "kernel-headers-{{ hostvars[inventory_hostname].ansible_kernel }}"
+  register: pkgheaders
+- name: get CentOS specific release version
+  slurp:
+    src: /etc/centos-release
+  register: os_release_file
+
+# If possible we use headers from main OS repo for this kernel.
+- name: Install Kernel Header and Devel from OS repo for Current Kernel
+  yum:
+    name:
+      - "kernel-headers-{{ hostvars[inventory_hostname].ansible_kernel }}"
+      - "kernel-devel-{{ hostvars[inventory_hostname].ansible_kernel }}"
+  when: pkgheaders.results|length > 0
+# CentOS VAULT - Kernel Headers not available in main OS repo anymore. Use Vault for this release
+- name: CentOS VAULT - Kernel Headers not available in main OS repo anymore. Use Vault for this release
+  # 2019-09-18: CentOS created new version 7.7.1908 and rebased the /7 repo.
+  # But vault for 7.6.1810 is not available but mirror is
+  when: pkgheaders.results|length == 0
+  block:
+    - name: ThisVault
+      yum_repository:
+        name: ThisVault
+        description: This Vault
+        baseurl: "{{ vault_repo_baseurl }}"
+        enabled: false
+    - name: ThisVault - Install Kernel Headers for Current Kernel
+      yum:
+        enablerepo: ThisVault
+        name: "kernel-headers-{{ hostvars[inventory_hostname].ansible_kernel | regex_search('\\d+\\.\\d+\\.\\d+-\\d+') }}.el{{ hostvars[inventory_hostname].ansible_distribution_major_version }}"
+    - name: ThisVault - Install Kernel Devel for Current Kernel
+      yum:
+        enablerepo: ThisVault
+        name: "kernel-devel-{{ hostvars[inventory_hostname].ansible_kernel | regex_search('\\d+\\.\\d+\\.\\d+-\\d+') }}.el{{ hostvars[inventory_hostname].ansible_distribution_major_version }}"
```

Comment:
Is `ThisVault` some commonly used name?

Reply:
there are multiple vaults for a single major version. So every centos minor version has its own vault. The repo is disabled by default so its should not interfere with something else.

Would `ThisCentOSVault` be a better matching name? I tried to avoid variable names for this making it a bit easier to debug

Reply:
I am ok with keeping the name. I was just curious.